File: ios/Sources/Core/Models/ImplementorInitRequest.swift
import Foundation

public struct ImplementorInitRequest: Codable, Equatable, Sendable {
    public let requestId: String
    public let implementorName: String
    public let implementorId: String
    public let implementorEphemeralPublicKey: String
    public let timestamp: Date
    public let expiresAt: Date

    enum CodingKeys: String, CodingKey {
        case requestId = "request_id"
        case implementorName = "implementor_name"
        case implementorId = "implementor_id"
        case implementorEphemeralPublicKey = "implementor_ephemeral_public_key"
        case timestamp
        case expiresAt = "expires_at"
    }

    public init(
        requestId: String,
        implementorName: String,
        implementorId: String,
        implementorEphemeralPublicKey: String,
        timestamp: Date,
        expiresAt: Date
    ) {
        self.requestId = requestId
        self.implementorName = implementorName
        self.implementorId = implementorId
        self.implementorEphemeralPublicKey = implementorEphemeralPublicKey
        self.timestamp = timestamp
        self.expiresAt = expiresAt
    }
}

public struct ClaimRequest: Codable, Equatable, Sendable {
    public let requestId: String
    public let claimCode: String
    public let deviceFingerprint: String
    public let deviceSignature: String

    enum CodingKeys: String, CodingKey {
        case requestId = "request_id"
        case claimCode = "claim_code"
        case deviceFingerprint = "device_fingerprint"
        case deviceSignature = "device_signature"
    }

    public init(requestId: String, claimCode: String, deviceFingerprint: String, deviceSignature: String) {
        self.requestId = requestId
        self.claimCode = claimCode
        self.deviceFingerprint = deviceFingerprint
        self.deviceSignature = deviceSignature
    }
}

public struct ClaimResponse: Codable, Equatable, Sendable {
    public let status: String
    public let pinnedToFingerprint: String

    enum CodingKeys: String, CodingKey {
        case status
        case pinnedToFingerprint = "pinned_to_fingerprint"
    }
}

public struct InitRespondRequest: Codable, Equatable, Sendable {
    public let requestId: String
    public let deviceFingerprint: String
    public let action: String
    public let encryptedMnemonic: String?
    public let rejectionReason: String?
    public let deviceSignature: String

    enum CodingKeys: String, CodingKey {
        case requestId = "request_id"
        case deviceFingerprint = "device_fingerprint"
        case action
        case encryptedMnemonic = "encrypted_mnemonic"
        case rejectionReason = "rejection_reason"
        case deviceSignature = "device_signature"
    }

    public init(
        requestId: String,
        deviceFingerprint: String,
        action: String,
        encryptedMnemonic: String? = nil,
        rejectionReason: String? = nil,
        deviceSignature: String
    ) {
        self.requestId = requestId
        self.deviceFingerprint = deviceFingerprint
        self.action = action
        self.encryptedMnemonic = encryptedMnemonic
        self.rejectionReason = rejectionReason
        self.deviceSignature = deviceSignature
    }
}

public struct InitRespondResponse: Codable, Equatable, Sendable {
    public let status: String
}

File: android/app/src/main/kotlin/com/wagmilabs/sigil/network/models/ImplementorInitRequest.kt
package com.wagmilabs.sigil.network.models

import com.google.gson.annotations.SerializedName
import java.util.Date

data class ImplementorInitRequest(
    @SerializedName("request_id")
    val requestId: String,
    @SerializedName("implementor_name")
    val implementorName: String,
    @SerializedName("implementor_id")
    val implementorId: String,
    @SerializedName("implementor_ephemeral_public_key")
    val implementorEphemeralPublicKey: String,
    @SerializedName("timestamp")
    val timestamp: Date,
    @SerializedName("expires_at")
    val expiresAt: Date
)

data class ClaimRequest(
    @SerializedName("request_id")
    val requestId: String,
    @SerializedName("claim_code")
    val claimCode: String,
    @SerializedName("device_fingerprint")
    val deviceFingerprint: String,
    @SerializedName("device_signature")
    val deviceSignature: String
)

data class ClaimResponse(
    @SerializedName("status")
    val status: String,
    @SerializedName("pinned_to_fingerprint")
    val pinnedToFingerprint: String
)

data class InitRespondRequest(
    @SerializedName("request_id")
    val requestId: String,
    @SerializedName("device_fingerprint")
    val deviceFingerprint: String,
    @SerializedName("action")
    val action: String,
    @SerializedName("encrypted_mnemonic")
    val encryptedMnemonic: String? = null,
    @SerializedName("rejection_reason")
    val rejectionReason: String? = null,
    @SerializedName("device_signature")
    val deviceSignature: String
)

data class InitRespondResponse(
    @SerializedName("status")
    val status: String
)
